src/main/java/com/vackosar/gitflowincrementalbuild/boundary/Configuration.java

```diff
@@ -62,6 +63,12 @@ private Configuration(MavenSession session) {
         Properties projectProperties = getProjectProperties(session);
         checkProperties(projectProperties);
 
+        if (Property.Constants.ALWAYS_MATCH_REGEX.equals(Property.enabledBranchRegex.getValue(projectProperties))) {
+            enabledBranchRegex = null;
```

Comment:
This should mapped as `Optional.empty()` instead.
TBH, I don't think we should reuse the same strategy from `exclude/includePathRegex`. I'd say it would be more logical if the property is not set all by default.
This would imply `disableIfBranchRegex` as the property name.

Reply:
renamed to `disableIfBranchRegex` as suggested.